src/ThriveBridgeDestination.sol

```diff
@@ -0,0 +1,196 @@
+    IThriveIERC20Wrapper public token;
+    mapping(address => mapping(uint256 => bool)) public mintNonces;
+    mapping(address => uint256) public burnNonces;
+
+    /**
+     * @dev Initializes the contract.
+     * @param _accessControlEnumerable The address of the AccessControlEnumerable contract.
+     * @param _role The access control role.
+     */
+    function initialize(
+        address _srcContract,
+        address _accessControlEnumerable,
+        bytes32 _role,
+        address _token
+    ) public initializer {
+        __Ownable_init(_msgSender());
+        __UUPSUpgradeable_init();
+
+        srcContract = _srcContract;
+        accessControlEnumerable =
+            IAccessControlEnumerable(_accessControlEnumerable);
+        role = _role;
+        token = IThriveIERC20Wrapper(_token);
+    }
+
+    /**
+     * @dev Overrides the authorization check for upgrading the contract implementation.
+     * Only the owner of this contract can authorize upgrades.
+     *
+     * @param newImplementation The address of the new implementation contract.
+     */
+    function _authorizeUpgrade(address newImplementation)
+        internal
+        override
+        onlyOwner
+    {}
+
+    /**
+     * @dev Modifier to only allow execution by admins.
+     * If the caller is not an admin, reverts with a corresponding message
+     */
+    modifier onlyAdmin() {
+        accessControlEnumerable.checkRole(role, _msgSender());
+        _;
+    }
+
+    /**
+     * @dev Sets the AccessControlEnumerable contract address.
+     * Only the owner of this contract can call this function.
+     *
+     * @param _accessControlEnumerable The address of the new AccessControlEnumerable contract.
+     * @param _role The new access control role.
+     */
+    function setAccessControlEnumerable(
+        address _accessControlEnumerable,
+        bytes32 _role
+    ) external onlyOwner {
+        accessControlEnumerable =
+            IAccessControlEnumerable(_accessControlEnumerable);
+        role = _role;
+    }
+
+    function mintTokens(
+        address sender,
+        address receiver,
+        uint256 amount,
+        uint256 nonce,
+        bytes calldata signature
+    ) external onlyAdmin {
+        _mintTokens(sender, receiver, amount, nonce, signature);
+    }
+
+    function _mintTokens(
+        address sender,
+        address receiver,
+        uint256 amount,
+        uint256 nonce,
+        bytes calldata signature
+    ) internal virtual nonReentrant {
+        require(amount > 0, "ThriveProtocol: amount must be greater than zero");
+
+        require(
+            !mintNonces[sender][nonce],
+            "ThriveProtocol: request already processed"
+        );
+        mintNonces[sender][nonce] = true;
+
+        bytes32 hash = SignatureHelper.hashBridgeRequest(
+            address(this), sender, receiver, nonce, amount
+        );
+        bool validSig =
+            SignatureHelper.verifyBridgeRequest(sender, hash, signature);
```

Comment:
can we use different name for this variable? In the first sight it did not look that understandable so maybe use `validSignature`